crates/bevy_pbr/src/light.rs

```diff
@@ -1226,16 +1397,29 @@ pub fn update_point_light_frusta(
         let view_translation = GlobalTransform::from_translation(transform.translation);
         let view_backward = transform.back();
 
-        for (view_rotation, frustum) in view_rotations.iter().zip(cubemap_frusta.iter_mut()) {
-            let view = view_translation * *view_rotation;
-            let view_projection = projection * view.compute_matrix().inverse();
+        if let Some((_inner_angle, angle)) = point_light.spotlight_angles {
+            let spot_view = spotlight_view_matrix(transform);
+            let spot_projection = spotlight_projection_matrix(angle);
+            let view_projection = spot_projection * spot_view.inverse();
 
-            *frustum = Frustum::from_view_projection(
+            cubemap_frusta.frusta[0] = Frustum::from_view_projection(
```

Comment:
it doesn't waste much resources (~80 bytes of cpu memory per shadow-casting spotlight) but yes it is ugly. i could change to use a vec of frusta for both, or use a separate component for the spotlight frustum. i would probably do a separate component if you don't have a strong preference.

Reply:
4 components in a Vec4 to define a Plane * 6 planes to define a frustum * 5 unused frusta = 480 bytes. But it's still not going to be much data until the number of shadow-mapped spotlights increases to any significant number. If doing a separate component is easy enough (I guess it would just be a separate bundle with a Frustum instead of CubemapFrusta?) then do that.

Reply:
added a separate SpotlightBundle with a SpotlightAngles component. it was a bit more complex than i thought, some plumbing had to be replicated to handle Frustum/VisibleEntities as opposed to CubemapFrusta/CubemapVisibleEntities. I think a separate bundle is good though.